src/Utility/Xml.php

```diff
@@ -307,7 +303,7 @@ public static function fromArray($input, array $options = [])
      * @return void
      * @throws \Cake\Utility\Exception\XmlException
      */
-    protected static function _fromArray(DOMDocument $dom, $node, &$data, $format): void
+    protected static function _fromArray(DOMDocument $dom, DOMDocument|DOMElement $node, $data, string $format): void
```

Comment:
I gave up trying to fix the $data type hint. We pass in many types even thought it only accept array.

Reply:
This whole Xml class needs to die :stuck_out_tongue:.

Reply:
It comes in handy when you need to use XML though. Being able to create an array out of XML and XML from an array can be useful at times.